bigquery/google/cloud/bigquery/client.py

```diff
@@ -205,6 +205,50 @@ def get_dataset(self, dataset_ref):
             method='GET', path=dataset_ref.path)
         return Dataset.from_api_repr(api_response, self)
 
+    def update_dataset(self, dataset, fields):
+        """Change some fields of a dataset.
+
+        Use ``fields`` to specify which fields to update. At least one field
+        must be provided. If a field is listed in ``fields`` and is ``None`` in
+        ``dataset``, it will be deleted.
+
+        If ``dataset.etag`` is not ``None``, the update will only
+        succeed if the dataset on the server has the same ETag. Thus
+        reading a dataset with ``get_dataset``, changing its fields,
+        and then passing it ``update_dataset`` will ensure that the changes
+        will only be saved if no modifications to the dataset occurred
+        since the read.
+
+        :type dataset: :class:`google.cloud.bigquery.dataset.Dataset`
+        :param dataset: the dataset to update.
+
+        :type fields: sequence of string
+        :param fields: the fields of ``dataset`` to change.
+
+        :rtype: :class:`google.cloud.bigquery.dataset.Dataset`
+        :returns: the modified ``Dataset`` instance
+        :raises: ValueError for fields that cannot be updated.
+
+        """
+        if dataset.project is None:
+            dataset._project = self.project
+        path = '/projects/%s/datasets/%s' % (dataset.project,
+                                             dataset.dataset_id)
+        partial = {}
+        for f in fields:
+            if f not in Dataset._updateable_fields:
```

Comment:
I might actually prefer a blacklist or no client-side check at all (let the backend handle issues).

I know that there will be fields that don't yet have support in the client libraries (alpha or we just haven't implemented them yet), and it will be nice if developers don't have to revert to full REST calls to use them.

Reply:
Okay, but I still need a dict from our client spelling (snake case) to the backend spelling (camel case), right? What should I do if something is not in the dict?

Or should I just transform from snake to camel?

Reply:
I think we can safely assume that the field names will always be camel case based on https://developers.google.com/protocol-buffers/docs/proto3#json "Message field names are mapped to lowerCamelCase and become JSON object keys."

We might eventually need a dictionary for backend names that are reserved keywords in Python, but I guess we'll address that when we get to it.

Reply:
Mapped snake to camel case.